front/src/reducers/osrdconf/helpers.ts

```diff
@@ -13,26 +14,44 @@ import type { PathStep } from './types';
 
 export const insertViaFromMap = (
   pathSteps: OperationalStudiesConfState['pathSteps'],
-  newVia: PathStep,
+  // newVia comes from a map click either on a track section or an operational point
+  newVia: Extract<PathStep, { track: string } | { uic: number }>,
   pathProperties: ManageTrainSchedulePathProperties
 ): OperationalStudiesConfState['pathSteps'] => {
+  const isOp = 'secondary_code' in newVia && 'uic' in newVia;
+
   // If one of these is missing, via is not valid (it hasn't been added via click on map) and we return the same array
-  if (!('track' in newVia) || !newVia.coordinates) return pathSteps;
+  if (!('track' in newVia) && !newVia.coordinates && !isOp) return pathSteps;
```

Comment:
I think the correct condition is this one: 
The pathStep is invalid if:
  - it is not an operational point AND it is not a track
  - OR it does not have coordinates

```suggestion
  if ((!isOp && !('track' in newVia)) || !newVia.coordinates) return pathSteps;
```

Reply:
Done